packages/plugin-ext/src/plugin/workspace.ts

```diff
@@ -169,4 +169,41 @@ export class WorkspaceExtImpl implements WorkspaceExt {
         return undefined;
     }
 
+    getWorkspaceFolder(uri: URI): theia.WorkspaceFolder | URI | undefined {
+        if (!this.folders || !this.folders.length) {
+            return undefined;
+        }
+
+        const resourcePath = uri.toString();
+
+        let workspaceFolder: theia.WorkspaceFolder | undefined;
+        for (let i = 0; i < this.folders.length; i++) {
+            const folder = this.folders[i];
+            const folderPath = folder.uri.toString();
+
+            if (resourcePath === folderPath) {
+                // return the input when the given uri is a workspace folder itself
+                return uri;
+            }
+
+            if (this.matchPaths(resourcePath, folderPath) && (!workspaceFolder || folderPath.length > workspaceFolder.uri.toString().length)) {
```

Comment:
You may go with something like this:
```typescript
 if (folderPath.startsWith(resourcePath) && (!workspaceFolder || folderPath.length > workspaceFolder.uri.toString().length)) {
```

Reply:
@mmorhun thanks for review
`startsWith` doesn't fit here, because in this case file `/home/user/project_2/src/main.js` matches the workspace folder `/home/user/project`, which is not correct.

Reply:
Well, you may ensure the trailing slash.
But up to you.